src/MailChimp/Facade.php

```diff
@@ -350,6 +350,43 @@ public function handle_grouping_integrity( $local_groups, $remote_groups, $group
 	}
 
 	/**
+	 * Create all segments necessary
+	 *
+	 * @param array $segments an array of segments
+	 * @return bool
+	 */
+	public function create_segments_from_groups( $segments ) {
+
+		foreach ( $segments as $segment ) {
+
+			try {
+
+				$result = $this->facade->lists->segmentTest( $this->current_list['id'], $segment );
+
+				if ( $result ) {
+
+					// Try adding the segment
+					$args = [
+						'type'         => 'saved',
+						'name'         => md5( $segment['conditions'][0]['value'] ), // 100 byte max so md5 it is
```

Comment:
Effectively we could use the overview page to show the linking between segment names and category!

I'm responding to your email right now. You have some good ideas there!